src/backend/distributed/planner/distributed_planner.c

```diff
@@ -163,13 +156,17 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 	{
 		if (IsLocalReferenceTableJoin(parse, rangeTableList))
 		{
+			List *referenceTableRTEList = NIL;
+			ExtractReferenceTableRTEList(rangeTableList, &referenceTableRTEList);
+
 			/*
 			 * For joins between reference tables and local tables, we replace
 			 * reference table names with shard tables names in the query, so
 			 * we can use the standard_planner for planning it locally.
 			 */
+			UpdateReferenceTablesWithShard(referenceTableRTEList);
```

Comment:
Perhaps the call to ExtractReferenceTableRTEList can be moved into UpdateReferenceTablesWithShard to make distributed_planner (the main entry point for Citus) a little cleaner?

Reply:
done